include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -93,125 +89,103 @@ void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
 
   const SizeType m = mat_c.nrTiles().rows();
   const SizeType n = mat_c.nrTiles().cols();
-  const SizeType mv = mat_v.nrTiles().rows();
-  const SizeType nv = mat_v.nrTiles().cols();
   const SizeType mb = mat_v.blockSize().rows();
-  const SizeType nb = mat_v.blockSize().cols();
   const SizeType ms = mat_v.size().rows();
-  const SizeType ns = mat_v.size().cols();
 
   // Matrix T
-  int tottaus;
-  if (ms < mb || ms == 0 || nv == 0)
-    tottaus = 0;
-  else
-    tottaus = (ms / mb - 1) * mb + ms % mb;
-
-  if (tottaus == 0)
+  if (m <= 1 || n == 0)
     return;
 
-  LocalElementSize sizeT(tottaus, tottaus);
-  TileElementSize blockSizeT(mb, mb);
-  Matrix<T, Device::CPU> mat_t(sizeT, blockSizeT);
+  const SizeType nr_reflector = mat_v.size().rows() - mb;
 
-  Matrix<T, Device::CPU> mat_vv({mat_v.size().rows(), mb}, mat_v.blockSize());
-  Matrix<T, Device::CPU> mat_w({mat_v.size().rows(), mb}, mat_v.blockSize());
-  Matrix<T, Device::CPU> mat_w2({mb, mat_c.size().cols()}, mat_c.blockSize());
+  dlaf::matrix::Distribution dist_t({nr_reflector, nr_reflector}, {mb, mb});
 
-  SizeType last_mb;
-  if (mat_v.blockSize().cols() == 1) {
-    last_mb = 1;
-  }
-  else if (mat_v.size().cols()) {
-    if (mat_v.size().cols() % mat_v.blockSize().cols() == 0)
-      last_mb = mat_v.blockSize().cols();
-    else
-      last_mb = mat_v.size().cols() % mat_v.blockSize().cols();
-  }
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsVV(n_workspaces,
+                                                                         mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsT(n_workspaces, dist_t);
+
+  SizeType last_mb = mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols();
 
   // Specific for V matrix layout where last column of tiles is empty
-  const SizeType last_reflector_idx = mat_v.nrTiles().cols() - 2;
+  const SizeType last_panel_reflector_idx = mat_v.nrTiles().cols() - 2;
+
+  for (SizeType k = last_panel_reflector_idx; k >= 0; --k) {

```

Comment:
This is a very minor thing, and it is totally subjective.


```suggestion
  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;

  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
```

IMHO it is more readable, because `-2` is less straightforward (at least to me) because it is the result of a `-1` because you want to skip the last panel plus `-1` because the index starts from 0.

In this way you are able to explicitly say "I'm computing the number of panels and I want to skip the last one" and then in the `for`, where it is clear that you deal with indices, the other `-1` is expected.